config/application.rb

```diff
@@ -78,6 +78,8 @@ class Application < Rails::Application
     # FIXME: required to make forms in pjax modals work
     config.action_controller.per_form_csrf_tokens = false
 
+    config.action_controller.forgery_protection_origin_check = false
+
```

Comment:
I think this is okay to do. And, we need it in order for CSRF validation to work.

When it comes to CORS, we'll probably have to enable rack-cors for production. We can lock it down to only allow our host, and only allow our host to access `/hmis-api` endpoints. Like this: https://github.com/greenriver/hmis-warehouse/pull/2258/commits/ffb9876bbce6f74ea143a573d1b1d87ddf67a9ce

For local development, it works fine without rack-cors. I'm not totally sure why, I think it might be because of this line https://github.com/greenriver/hmis-warehouse/blob/5aaf4bf38427712c891f352aa9172c071b52993f/config/environments/development.rb#L17

Reply:
@yakloinsteak or @ttoomey do either of you know if this is safe?

Reply:
If the new front end is on a different domain, this might be required to be false, but if `config.hosts` includes the new front-end domain, we might be able to set it to true, which would be preferred. I'm not sure reading the docs exactly how it behaves.

Reply:
Sorry, I think I muddled things by writing about CORS in my original comment. The rack-cors setup and and `config.host` setting are not directly related to this config, which is strictly about how CSRF validation works. The definition for [forgery_protection_origin_check](https://edgeguides.rubyonrails.org/configuring.html#config-action-controller-forgery-protection-origin-check) is:
> Configures whether the HTTP Origin header should be checked against the site's origin as an additional CSRF defense.

So this configures whether the `verify_authenticity_token` action rejects CSRF tokens that don't match the site origin. We're still getting CSRF protection here, we're just turning off the extra origin check. During _development_, this needs to be set to `false` (regardless of how `config.hosts` is set) because the frontend has a different origin. Now that I'm giving it some more thought, though, I'm pretty sure we can  reverse-proxy our way out of this when its deployed. Given that, I'll move this configuration to `development.rb` and hide it behind the `ENABLE_HMIS_API` flag.

Reply:
Thanks.  That sounds much cleaner to me.